crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -175,7 +315,7 @@ async fn handle_set_code(
 async fn handle_set_nonce(state: StateType, address: Address, nonce: U256) -> ResponseData<()> {
     match TryInto::<u64>::try_into(nonce) {
         Ok(nonce) => {
-            match (*state).write().await.modify_account(
+            match (*state.rethnet).write().await.modify_account(
```

Comment:
Can't you do `state.rethnet.write()` here too?

Reply:
i think you're looking at an old/outdated revision? as explained [here](https://github.com/NomicFoundation/hardhat/pull/3998#discussion_r1236001000), i made these changes across two different commits (one in the "block spec handling" commit, and another specific to the remaining instances that didn't have anything to do with block spec handling).